numpy/random/randomgen/generator.pyx

```diff
@@ -0,0 +1,4149 @@
+    accessed using ``MT19937``. ``RandomGenerator``, besides being
+    NumPy-aware, has the advantage that it provides a much larger number
+    of probability distributions to choose from.
+
+    Examples
+    --------
+    >>> from np.random.randomgen import RandomGenerator
+    >>> rg = RandomGenerator()
+    >>> rg.standard_normal()
+
+    Using a specific generator
+
+    >>> from np.random.randomgen import MT19937
+    >>> rg = RandomGenerator(MT19937())
+
+    The generator is also directly available from basic RNGs
+
+    >>> rg = MT19937().generator
+    >>> rg.standard_normal()
+    """
+    cdef public object _basicrng
+    cdef brng_t *_brng
+    cdef binomial_t *_binomial
+    cdef object lock
+    poisson_lam_max = POISSON_LAM_MAX
+
+    def __init__(self, brng=None):
+        if brng is None:
+            brng = Xoroshiro128()
+        self._basicrng = brng
+
+        capsule = brng.capsule
+        cdef const char *name = "BasicRNG"
+        if not PyCapsule_IsValid(capsule, name):
+            raise ValueError("Invalid brng. The brng must be instantized.")
+        self._brng = <brng_t *> PyCapsule_GetPointer(capsule, name)
+        self._binomial = <binomial_t *>malloc(sizeof(binomial_t))
+        self.lock = brng.lock
+
+    def __dealloc__(self):
+        free(self._binomial)
+
+    def __repr__(self):
+        return self.__str__() + ' at 0x{:X}'.format(id(self))
+
+    def __str__(self):
+        _str = self.__class__.__name__
+        _str += '(' + self._basicrng.__class__.__name__ + ')'
+        return _str
+
+    # Pickling support:
+    def __getstate__(self):
+        return self.state
+
+    def __setstate__(self, state):
+        self.state = state
+
+    def __reduce__(self):
+        from ._pickle import __generator_ctor
+        return (__generator_ctor,
+                (self.state['brng'],),
+                self.state)
+
+    def seed(self, *args, **kwargs):
+        """
+        Reseed the basic RNG.
+
+        Parameters depend on the basic RNG used.
+
+        Notes
+        -----
+        Arguments are directly passed to the basic RNG. This is a convenience
+        function.
+
+        The best method to access seed is to directly use a basic RNG instance.
+        This example demonstrates this best practice.
+
+        >>> from numpy.random.randomgen import RandomGenerator, PCG64
+        >>> brng = PCG64(1234567891011)
+        >>> rg = brng.generator
+        >>> brng.seed(1110987654321)
+
+        The method used to create the generator is not important.
+
+        >>> brng = PCG64(1234567891011)
+        >>> rg = RandomGenerator(brng)
+        >>> brng.seed(1110987654321)
+
+        These best practice examples are equivalent to
+
+        >>> rg = RandomGenerator(PCG64(1234567891011))
+        >>> rg.seed(1110987654321)
+
+        """
+        # TODO: Should this remain
+        self._basicrng.seed(*args, **kwargs)
+        return self
```

Comment:
I think we should leave this in legacy but remove it here. If needed, we can always add it.

Reply:
This can be dropped but then `basicrng` needs to stop being private so there is a decent way to reseed.

Now
```
rg = RandomGenerator()
rg.seed(1234)
```

If dropped
```
rg = RandomGenerator()
rg._basicrng.seed(1234)
```

Same goes for state setting and getting.